dali/operators/image/convolution/laplacian.cc

```diff
@@ -0,0 +1,222 @@
+      output_desc[0].shape.set_tensor_shape(sample_idx, input[sample_idx].shape());
+    }
+    return true;
+  }
+
+  void RunImpl(workspace_t<CPUBackend>& ws) override {
+    const auto& input = ws.template Input<CPUBackend>(0);
+    auto& output = ws.template Output<CPUBackend>(0);
+    output.SetLayout(input.GetLayout());
+    auto in_shape = input.shape();
+    auto& thread_pool = ws.GetThreadPool();
+
+    int nsamples = input.shape().num_samples();
+    for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
+      const auto& shape = input[sample_idx].shape();
+      auto elem_volume = volume(shape.begin() + dim_desc_.usable_axes_start, shape.end());
+      auto priority = elem_volume * args.GetTotalWindowSizes(sample_idx);
+
+      int seq_elements = 1;
+      int64_t stride = 0;
+      if (dim_desc_.is_sequence) {
+        seq_elements = volume(shape.begin(), shape.begin() + dim_desc_.usable_axes_start);
+        stride = elem_volume;
+      }
+      for (int elem_idx = 0; elem_idx < seq_elements; elem_idx++) {
+        thread_pool.AddWork(
+            [this, &input, &output, sample_idx, elem_idx, stride](int thread_id) {
+              const auto& scales = args.GetScales(sample_idx);
+              auto elem_shape = input[sample_idx].shape().template last<ndim>();
+              auto in_view = TensorView<StorageCPU, const In, ndim>{
+                  input[sample_idx].template data<In>() + stride * elem_idx, elem_shape};
+              auto out_view = TensorView<StorageCPU, Out, ndim>{
+                  output[sample_idx].template mutable_data<Out>() + stride * elem_idx, elem_shape};
+              // Copy context so that the kernel instance can modify scratchpad
+              auto ctx = ctx_;
+              kmgr_.Run<Kernel>(thread_id, sample_idx, ctx, out_view, in_view, windows_[sample_idx],
+                                scales);
+            },
+            priority);
+      }
+    }
+    thread_pool.RunAll();
+  }
+
+ private:
+  LaplacianArgs<axes> args;
+  DimDesc dim_desc_;
+
+  kernels::KernelManager kmgr_;
+  kernels::KernelContext ctx_;
+
+  LaplacianWindows<float> lap_windows_;
+  std::vector<std::array<std::array<TensorView<StorageCPU, const float, 1>, axes>, axes>> windows_;
+};
+
+
+}  // namespace laplacian
+
+bool Laplacian::SetupImpl(std::vector<OutputDesc>& output_desc, const workspace_t<CPUBackend>& ws) {
+  const auto& input = ws.template Input<CPUBackend>(0);
+  auto layout = input.GetLayout();
+  auto dim_desc = ParseAndValidateDim(input.shape().sample_dim(), layout);
+  dtype_ = dtype_ != DALI_NO_TYPE ? dtype_ : input.type();
+  DALI_ENFORCE(dtype_ == input.type() || dtype_ == DALI_FLOAT,
+               "Output data type must be same as input, FLOAT or skipped (defaults to input type)");
+
+  if (!impl_ || impl_in_dtype_ != input.type() || impl_dim_desc_ != dim_desc) {
+    impl_in_dtype_ = input.type();
+    impl_dim_desc_ = dim_desc;
+
+    // clang-format off
```

Comment:
Do we want to have this kind of annotation in the code?

Reply:
We do have it already, but probably I can remove it.

Reply:
TYPE_SWITCH should be clang-format immune for quite some time already.

Reply:
done